core/blockchain.go

```diff
@@ -2455,6 +2466,23 @@ func (bc *BlockChain) reportBlock(block *types.Block, res *ProcessResult, err er
 	log.Error(summarizeBadBlock(block, receipts, bc.Config(), err))
 }
 
+// logForkReadiness will write a log when a future fork is scheduled, but not
+// active. This is useful so operators know their client is ready for the fork.
+func (bc *BlockChain) logForkReadiness(block *types.Block) {
+	c := bc.Config()
+	current, last := c.LatestFork(block.Time()), c.LatestFork(math.MaxUint64)
+	t := c.Timestamp(last)
+	if t == nil {
+		return
+	}
+	at := time.Unix(int64(*t), 0)
+	if current < last && time.Now().After(bc.lastForkReadyAlert.Add(forkReadyInterval)) {
+		log.Info("Ready for fork activation", "fork", last, "date", at.Format(time.RFC822),
+			"remaining", time.Until(at).Round(time.Second), "timestamp", at.Unix())
+		bc.lastForkReadyAlert = time.Now()
+	}
+}
+
```

Comment:
Why not check _if_ we should log out before we compute the ready activation?

```suggestion
// logForkReadiness will write a log when a future fork is scheduled, but not
// active. This is useful so operators know their client is ready for the fork.
func (bc *BlockChain) logForkReadiness(block *types.Block) {
	if time.Since(bc.lastForkReadyAlert) > forkReadyInterval {
		c := bc.Config()
		current, last := c.LatestFork(block.Time()), c.LatestFork(math.MaxUint64)
		t := c.Timestamp(last)
		if t == nil {
			return
		}
		at := time.Unix(int64(*t), 0)
		if current < last {
			log.Info("Ready for fork activation", "fork", last, "date", at.Format(time.RFC822),
				"remaining", time.Until(at).Round(time.Second), "timestamp", at.Unix())
			bc.lastForkReadyAlert = time.Now()
		}
	}
}
```

Reply:
Ah nvm, we're never updating bc.lastForkReadyAlert this way